subt/cloudsim2osgar.py

```diff
@@ -404,35 +408,46 @@ def convert_depth(self, msg):
         # Pose of the robot relative to starting position.
         try:
-            self.tf.waitForTransform('global', self.robot_name, msg.header.stamp, rospy.Duration(0.3))
-            robot_pose = self.tf.lookupTransform('global', self.robot_name, msg.header.stamp)
+            self.tf.waitForTransform('global', self.robot_name, timestamp, rospy.Duration(0.3))
+            robot_pose = self.tf.lookupTransform('global', self.robot_name, timestamp)
         except (tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException, tf.Exception) as e:
             rospy.logerr('tf global error: {}'.format(e))
             return None
         # Position of the camera relative to the robot.
         try:
-            # RTABMAP produces rotation in a visual coordinate frame (Z axis
-            # forward), but we need to match the world frame (X axis forward)
-            # of robot_pose.
-            WORLD_TO_OPTICAL = tf.transformations.quaternion_from_matrix(
-                    [[ 0, -1,  0, 0],
-                     [ 0,  0, -1, 0],
-                     [ 1,  0,  0, 0],
-                     [ 0,  0,  0, 1]])
-            self.tf.waitForTransform(self.robot_name, msg.header.frame_id, msg.header.stamp, rospy.Duration(0.3))
+            self.tf.waitForTransform(self.robot_name, frame_id, timestamp, rospy.Duration(0.3))
             camera_xyz, camera_quat = self.tf.lookupTransform(
-                    self.robot_name, msg.header.frame_id, msg.header.stamp)
-            camera_pose = (camera_xyz,
-                    tf.transformations.quaternion_multiply(camera_quat, WORLD_TO_OPTICAL).tolist())
+                    self.robot_name, frame_id, timestamp)
+            if is_optical_frame:
+                # RTABMAP produces rotation in a visual coordinate frame (Z axis
+                # forward), but we need to match the world frame (X axis forward)
+                # of robot_pose.
+                WORLD_TO_OPTICAL = tf.transformations.quaternion_from_matrix(
+                        [[ 0, -1,  0, 0],
+                         [ 0,  0, -1, 0],
+                         [ 1,  0,  0, 0],
+                         [ 0,  0,  0, 1]])
+                camera_pose = (camera_xyz,
+                        tf.transformations.quaternion_multiply(camera_quat, WORLD_TO_OPTICAL).tolist())
+            else:
+                camera_pose = camera_xyz, camera_quat
         except (tf.LookupException, tf.ConnectivityException, tf.ExtrapolationException, tf.Exception) as e:
             rospy.logerr('tf camera error: {}'.format(e))
             return None
+
+        return robot_pose, camera_pose
+
+
+    def convert_rgbd(self, msg):
+        cam = self.camera_poses(msg.header.stamp, msg.header.frame_id)
+        if cam is None:
+            return None
         rgb = msg.rgb_compressed.data
         d = msg.depth_compressed.data
-        return (robot_pose, camera_pose, rgb, d)
+        return cam + (rgb, d)
```

Comment:
I am not sure if to address the tuple vs. list issue (msgpack) here if `cloudsim2osgar` could not be replayed?